mne/preprocessing/maxwell.py

```diff
@@ -0,0 +1,489 @@
+# Authors: Mark Wronkiewicz <wronk.mark@gmail.com>
+#          Jussi Nurminen <jnu@iki.fi>
+
+# This code was adapted and relicensed (with BSD form) with permission from
+# Jussi Nurminen
+
+# License: BSD (3-clause)
+
+# Note, there are an absurd number of different possible notations for
+# spherical coordinates, which confounds the notation for spherical harmonics.
+# Here, we purposefully stay away from shorthand notation in both and use
+# explicit terms (like 'azimuth' and 'polar') to avoid confusion.
+
+# TODO: write in equation numbers from Samu's paper
+
+import numpy as np
+from scipy.special import lpmv
+from scipy.linalg import pinv
+from ..forward._compute_forward import _concatenate_coils
+#from ..fixes import partial
+from scipy.misc import factorial as fact
+from ..io.constants import FIFF
+from .. import pick_types, pick_info
+from ..utils import logger
+from ..forward._make_forward import _read_coil_defs, _create_coils
+
+
+def maxwell_filter(raw, coils, origin, int_order=8, ext_order=3, n_jobs=1):
+    """Apply Maxwell filter to data using spherical harmonics.
+
+    Parameters
+    ----------
+    raw : instance of mne.io.Raw
+        Data to be filtered
+    coils : list
+        List of MEG coils. Each element must a contain coil information dict
+        continaining 'rmag', 'cosmag', and 'w' keys
```

Comment:
That sounds right. The best we have is the accurate / 16 points.

@staulu can get the definitions for these other points? Are they available
or published somewhere? We probably want them, at least for this
application, if not also more generally e.g. in forward generation.
@agramfort do you agree?

@mshamalainen any comments on this third / alternate set of coil
integration points that Elekta uses?